Features/native support for rq 1.0+

coolhacker opened the pull request:
The first iteration, there may be a thousand bugs.
P.S.: What do you think about the idea to rework blueprints to \<queue>\<queue_registry> form?

coolhacker committed:
Edit travis config to not test RQ <1.0

eoranged commented on rq_dashboard/web.py:
```diff
@@ -267,17 +240,24 @@ def list_instances():
 def list_jobs(queue_name, page):
     current_page = int(page)
-    queue = get_queue(queue_name)
+    registry_swapper = '-failed' in queue_name
```
Better to implement as `/jobs/<queue_name>/registries/<registry_name>/<page>.json`

coolhacker replied:
Implemented.

eoranged commented on rq_dashboard/web.py:
```diff
@@ -164,23 +144,16 @@ def pagination_window(total_items, cur_page, per_page=5, window_size=10):
 @blueprint.route('/<queue_name>', defaults={'page': '1'})
 @blueprint.route('/<queue_name>/<page>')
```
`@blueprint.route('/<queue_name>/registries/<registry_name>')`
`@blueprint.route('/<queue_name>/registries/<registry_name>/<page>')`

coolhacker replied:
Added.

coolhacker committed:
Rework tests for RQ 1.0+

eoranged commented on rq_dashboard/web.py:
```diff
@@ -160,27 +142,44 @@ def pagination_window(total_items, cur_page, per_page=5, window_size=10):
+
+
+@blueprint.route('/', defaults={'queue_name': None, 'registry_name': 'queued', 'page': '1'})
+@blueprint.route('/<queue_name>', defaults={'registry_name': 'queued', 'page': '1'})
```
trailing slash please

eoranged replied:
Also, It might make sense to use something like `/queues/<queue_name>/...`, but let's do this separately.

eoranged commented on rq_dashboard/web.py:
```diff
@@ -28,25 +28,13 @@
 from rq import (Queue, Worker, cancel_job, pop_connection,
                 push_connection, requeue_job, VERSION as rq_version)
+
+from rq.registry import FailedJobRegistry, StartedJobRegistry, FinishedJobRegistry, DeferredJobRegistry
```
please use isort

coolhacker committed:
First iteration of reworking for RQ 1.0+

Delete compat.py, rework web.py for native working with RQ 1.0+
Rework blueprints for failed jobs and classic view for queue


Fix


Some small fixes


Rename classic view to queued jobs

In rq lib jobs may be a queued, started, finished, failed and deferred
fix


Rework


fix


rework buttons for working with different registries


yeah


Implement isort

somefix
